rfcs/0000-wildcard-data-type.md

```diff
@@ -0,0 +1,150 @@
+
+## Fields
+
+<!--
+Stage: 1: Describe at a high level how this change affects fields. Which fieldsets will be impacted? How many fields overall? Are we primarily adding fields, removing fields, or changing existing fields? The goal here is to understand the fundamental technical implications and likely extent of these changes. ~2-5 sentences.
+-->
+
+Adopting wildcard in a field will require changing the defined `type` in the schema from `keyword` to `wildcard`. One example is provided for this initial proposal as an example; a later stage proposal will include the entire list of proposed fields.
+
+Current definition as of ECS 1.5.0:
+
+```yaml
+    - name: command_line
+      level: extended
+      type: keyword
+      short: Full command line that started the process.
+...
+      multi_fields:
+      - type: text
+        name: text
+```
+
+Proposed type change:
+
+```yaml
+    - name: command_line
+      level: extended
+      type: wildcard
+      short: Full command line that started the process.
+...
+      multi_fields:
+      - type: text
+        name: text
+```
+
+Note the existing `text` data type multi-field will remain to support tokenized searches.
+
+## Usage
+
+<!--
+Stage 1: Describe at a high-level how these field changes will be used in practice. Real world examples are encouraged. The goal here is to understand how people would leverage these fields to gain insights or solve problems. ~1-3 paragraphs.
+-->
+
+Many use cases benefit from wildcard search performance across long unstructured or semi-structured fields. Particularly in the security threat hunting and detection disciplines, reliable and efficient wildcard usage is a requirement from users. The following
+sections list common use cases which would benefit from improved wildcard searching.
+
+#### Paths and names
+
+Searching across various permutations of various file system and URL structures to find a particular file, path, or other resource. Leading wildcard searches on these fields may be necessary to users, but (as keywords) are known to produce very poor search performance
+(e.g. let's search for all instances of a known file without an unknown path: `file.path:*syslog.log`).
+
+Example data categories:
+
+* File paths
+* URLs
+* Usernames
+* User agent
+* Process names
+* Email addresses
+* OS names
+* Domain names
+* Host names
+* Registry data
+
+#### Stack traces
+
+Program stack traces tend to be well-structured but with long text and varied contents. There are too many subtleties and application-specific patterns to map all of them accurately with ECS' field definitions. Better performing wildcard searches can help the user formulate their own queries easier and with a smaller performance hit.
+
+### Command-line execution
+
+Full command-line events often have a series of options or flags which may appear in any arbitrary order. Wildcard searches are often required to successfully identify events regardless of the ordering.
```

Comment:
Actually to work around the ordering of the flags, using the array of keywords `process.args` is much better.

The power of having `wildcard` on `process.command_line` is instead to detect subtle patterns like multiple spaces, detect specific substrings, or specifically to take advantage of wildcard's respect of the order to filter for a flag and a specific argument passed to that flag.

Reply:
Thanks this is good context. I've updated my wording to better capture this detail.